lib/runtime-core/src/memory/dynamic.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{
+    memory::{WASM_MAX_PAGES, WASM_PAGE_SIZE},
+    sys,
+    types::MemoryDesc,
+    vm,
+};
+
+pub const DYNAMIC_GUARD_SIZE: usize = 4096;
+
```

Comment:
Could we add some documentation describing `DynamicMemory`, `StaticMemory`, `SharedStaticMemory`, their differences and usages.